UI/src/main/java/adList/AdList.java

```diff
@@ -0,0 +1,85 @@
+package adList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AdList {
+    private List<Ad> adList;
+
+    public AdList(){
+        adList = new ArrayList<>();
+    }
+    public AdList(List<Ad> adList){
+        this.adList = new ArrayList<>();
+        adList.forEach(ad -> this.adList.add(ad.clone()));
+    }
+    public List<Ad> getPage(int skip, int top){
+        return getPage(skip, top,null);
+    }
+    public List<Ad> getPage(int skip, int top, FilterConfig filterConfig){
+
+        List<Ad> result = new ArrayList<>();
+        if(filterConfig == null){
+            result = adList.subList(skip, Math.min(top + skip, adList.size()));
+        } else {
+            for (Ad ad : adList) {
+                if (filterConfig.checkAd(ad)) {
+                    result.add(ad);
+                }
+            }
+            result = result.subList(skip, Math.min(top + skip, adList.size()));
+        }
+        return result;
+    }
+    public Ad get(String id){
+        if(adList == null){
+            return null;
+        }
+        List<Ad> ads = adList.stream().filter(ad -> ad.getId().equals(id)).collect(Collectors.toList());
+        return ads.size() != 0 ? ads.get(0) : null;
+    }
+    public boolean add(Ad adItem) {
+        if(AdList.validate(adItem) && get(adItem.getId()) == null) {
+            return adList.add(adItem.clone());
+        } else {
+            return false;
+        }
+    }
+    public boolean edit(String id, AdIncomplete adItem){
+        Ad ad = this.get(id);
+        if(ad == null){
+            return false;
+        } else {
```

Comment:
`else` is redundant after `return`. You can just continue your logic starting from line 14. The fact that we are here means there has not been a return yet, therefore `ad != null` by all means